fusionbox/fabric/django/new.py

```diff
@@ -143,18 +144,25 @@ def generate_pyc():
         run('python -m compileall . > /dev/null')
 
 
+@contextlib.contextmanager
+def cd_git_extract(gitref, tmp_dir=tempfile.mkdtemp()):
+    # last argument adds trailing slash, which is needed by rsync
+    extract_dir = os.path.join(tmp_dir, gitref, '')
+
+    if not os.path.exists(extract_dir):
+        local('git archive {ref} | tar x -C {dir}'.format(ref=gitref, dir=extract_dir))
+        # removes the tmpdir, other registered functions will do nothing
+        atexit.register(shutil.rmtree, tmp_dir, ignore_errors=True)
+
+    with lcd(extract_dir):
+        yield extract_dir
```

Comment:
I *think* this will lcd just for the time it takes to yield the extract_dir and by the time any code is executed afterward, the context manager would have been exited. I would prefer it if this context manager (`cd_git_extract`) managed the cwd more explicitly.

Reply:
I think this works; see https://github.com/fusionbox/fusionbox-fabric-helpers/blob/639b55718db1a1f07a4be12eb0a5945e22307057/fusionbox/fabric/django/new.py#L42